### PR TITLE
[SPARK-36480][SS] SessionWindowStateStoreSaveExec should not filter input rows against watermark

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This PR proposes to remove the filter applying to input rows against watermark in SessionWindowStateStoreSaveExec, since SessionWindowStateStoreSaveExec is expected to store all inputs into state store, and apply eviction later.

### Why are the changes needed?

The code is logically not right, though I can't reproduce the actual problem.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests. I can't come up with broken case failing on previous code, but we can review the logic instead.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala [hunk 1]
```diff
@@ -642,9 +647,7 @@ case class SessionWindowStateStoreSaveExec(
         // Assumption: watermark predicates must be non-empty if append mode is allowed
         case Some(Append) =>
           allUpdatesTimeMs += timeTakenMs {
-            val filteredIter = applyRemovingRowsOlderThanWatermark(iter,
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala:645]: Hmm, I think rows older than watermark are dropped at earlier operators? This looks like a no-op?

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala:645]: Oh yea, actually in the below iterator, we remove the values by watermark as condition. It is same as here.

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thanks for fixing this. Merging to master/3.2.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala (hunk 1 above)
Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala:645]: The difference is that here we only filtered out sessions which new events are appeared for the grouping keys, whereas below we filtered out sessions from all state rows.

To trigger the problem, the grouping key should have existing sessions which has the end time earlier than the watermark. It's a bit tricky based of how watermark works in SS and how no-data batch is triggered. These sections are expected to be evicted on no-data batch so tricky to reproduce.

**Comment by HeartSaVioR** [comment]
Thanks @viirya for the quick reviewing and merging!

**Pull request closed without merging**